Packages/com.chisel.editor/Chisel/Editor/Components/Generators/CSGBox/Editor/CSGBoxGeneratorMode.cs

```diff
@@ -76,19 +78,20 @@ public void OnSceneGUI(SceneView sceneView, Rect dragArea)
                     CSGEditModeManager.EditMode = CSGEditMode.ShapeEdit;
                     break;
                 }
+
                 case BoxExtrusionState.Cancel:
                 {
-                    if (box)
-                        UnityEngine.Object.DestroyImmediate(box.gameObject);
```

Comment:
When you create an object in an empty scene, it'll create a model for you and place it underneath it.
But when canceling creating a box by pressing escape, this would only destroy the box.

Reply:
This isn't related to the PR but reminded me i'm occasionally getting situations where new objects are always created in new models, not any existing one.

Reply:
Yeah I noticed that too .. I'll make a ticket for it

Reply:
Created issue #65 for this